### ISSUE TITLE
Incorrect overflow-flag when subtracting from a negative number

### ISSUE THREAD

**Issue by kjonas65** [kirkedal/compSys-e2017-sim#17]
The following test will not follow the jump, even though -2 is clearly less than 0
```
	movq $0, %rax
	movq $-2, %rbx
	cmpq %rax, %rbx
	jl label
	nop
label:	nop
```
As far as I can tell this is caused by the overflow-flag incorrectly being set.

**Comment by madsobitsoe** [comment]
Description of issue:  
The following asm illustrates the OF being set wrong.
```
    movq $0, %rax
    movq $-2, %rbx
    cmpq %rax, %rbx
    jl label
    nop
label:    nop
```


`cmpq %rax, %rbx` run on the sim, will set flags to:
- `ZF=0`
- `SF=1`
- `OF=1`

The flags should instead be set to:
- `ZF=0`
- `SF=1`
- `OF=0`


According to this site:

https://en.wikibooks.org/wiki/X86_Assembly/X86_Architecture#EFLAGS_Register

the `OF` flag will be
> Set if signed arithmetic operations result in a value too large for the register to contain.

**Comment by finnschiermer** [comment]
The error is in https://github.com/kirkedal/compSys-e2017-sim/blob/master/architecture-tools/sim_step.c#L288. The arguments to computeFlags are reversed.

**Comment by finnschiermer** [comment]
Thx for reporting this, I'll fix it, the proposed fix in #18 will work, but I want it fixed in a different way.

**Comment by finnschiermer** [comment]
@madsobitsoe @kjonas65 Sorry for the long wait.
